File: 00-每日一题/20200508_129.cpp
//
// Created by zanbo on 2020/5/8.
//

class Solution {
public:
    int sum=0;
    int sumNumbers(TreeNode* root) {
        if(root==NULL) return sum;
        helper(root,"");
        return sum;
    }

    void helper(TreeNode* node,string temp)
    {
        if(node==NULL) return;
        temp+= to_string(node->val);
        if(node->right==NULL && node->left==NULL)
        {
            int inttemp = atoi(temp.c_str());
            sum += inttemp;
            return;
        }
        helper(node->left,temp);
        helper(node->right,temp);
        return;
    }
};

File: 00-每日一题/20200508_129.py
# 求根到叶子节点数字之和

class Solution(object):
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root is None:
            return 0
        res = []
        self._dfs(root, 0, res)
        return sum(res)

    def _dfs(self, root, pre_v, res):

        if root.left is None and root.right is None:
            # 直接把前面的结果乘以10
            res.append(pre_v*10+root.val)
        if root.left:
            self._dfs(root.left, pre_v*10+root.val, res)
        if root.right:
            self._dfs(root.right, pre_v*10+root.val, res)